File: webclient/ts/API/passwords.ts
import zxcvbn from "zxcvbn";

const commonWords = ["aurum", "finitum"];

export const verifyPassword = async (password: string, userInput: string[]): Promise<zxcvbn.ZXCVBNResult | string> => {
    const res = zxcvbn(password, commonWords.concat(...userInput));

    if (res.score < 2) {
        return res;
    }

    if (password.length < 8) {
        return "Password too short";
    }

    if (password.length > 72) {
        return "Password too long";
    }

    return res;
};

File: internal/passwords/passwords.go
package passwords

import (
	"github.com/trustelem/zxcvbn"
)

var commonTerms = []string{
	"aurum",
	"finitum",
}

func CheckStrength(password string, userinput []string) bool {
	if len(password) < 8 {
		return false
	}

	// Max length for bcrypt
	if len(password) > 72 {
		return false
	}

	disallowed := append(commonTerms, userinput...)
	res := zxcvbn.PasswordStrength(password, disallowed)

	return res.Score > 2
}
